server/opts.go

```diff
@@ -615,6 +660,206 @@ func parseCluster(v interface{}, opts *Options, errors *[]error, warnings *[]err
+				continue
+			}
+			o.Gateway.Host = hp.host
+			o.Gateway.Port = hp.port
+		case "port":
+			o.Gateway.Port = int(mv.(int64))
+		case "host", "net":
+			o.Gateway.Host = mv.(string)
+		case "authorization":
+			auth, err := parseAuthorization(tk, o, errors, warnings)
+			if err != nil {
+				*errors = append(*errors, err)
+				continue
+			}
+			if auth.users != nil {
+				*errors = append(*errors, &configErr{tk, "Gateway authorization does not allow multiple users"})
+				continue
+			}
+			o.Gateway.Username = auth.user
+			o.Gateway.Password = auth.pass
+			o.Gateway.AuthTimeout = auth.timeout
+		case "tls":
+			config, timeout, err := getTLSConfig(tk)
+			if err != nil {
+				*errors = append(*errors, err)
+				continue
+			}
+			o.Gateway.TLSConfig = config
+			o.Gateway.TLSTimeout = timeout
+		case "advertise":
+			o.Gateway.Advertise = mv.(string)
+		case "connect_retries":
+			o.Gateway.ConnectRetries = int(mv.(int64))
+		case "default_permissions":
+			perms, err := parseGatewayPermissions(mv, errors, warnings)
+			if err != nil {
+				*errors = append(*errors, err)
+				continue
+			}
+			o.Gateway.DefaultPermissions = perms
+		case "gateways":
+			gateways, err := parseGateways(mv, errors, warnings)
+			if err != nil {
+				return err
+			}
+			o.Gateway.Gateways = gateways
+		case "reject_unknown":
+			o.Gateway.RejectUnknown = mv.(bool)
+		default:
+			if !tk.IsUsedVariable() {
+				err := &unknownConfigFieldErr{
+					field: mk,
+					configErr: configErr{
+						token: tk,
+					},
+				}
+				*errors = append(*errors, err)
+				continue
+			}
+		}
+	}
+	return nil
+}
+
+// Parse TLS and returns a TLSConfig and TLSTimeout.
+// Used by cluster and gateway parsing.
+func getTLSConfig(tk token) (*tls.Config, float64, error) {
+	tc, err := parseTLS(tk)
+	if err != nil {
+		return nil, 0, err
+	}
+	config, err := GenTLSConfig(tc)
+	if err != nil {
+		err := &configErr{tk, err.Error()}
+		return nil, 0, err
+	}
+	// For clusters/gateways, we will force strict verification. We also act
+	// as both client and server, so will mirror the rootCA to the
+	// clientCA pool.
+	config.ClientAuth = tls.RequireAndVerifyClientCert
+	config.RootCAs = config.ClientCAs
+	return config, tc.Timeout, nil
+}
+
+func parseGateways(v interface{}, errors *[]error, warnings *[]error) ([]*RemoteGatewayOpts, error) {
+	tk, v := unwrapValue(v)
+	// Make sure we have an array
+	ga, ok := v.([]interface{})
+	if !ok {
+		return nil, &configErr{tk, fmt.Sprintf("Expected gateways field to be an array, got %T", v)}
+	}
+	gateways := []*RemoteGatewayOpts{}
+	for _, g := range ga {
+		tk, g = unwrapValue(g)
+		// Check its a map/struct
+		gm, ok := g.(map[string]interface{})
+		if !ok {
+			*errors = append(*errors, &configErr{tk, fmt.Sprintf("Expected gateway entry to be a map/struct, got %v", g)})
+			continue
+		}
+		gateway := &RemoteGatewayOpts{}
+		for k, v := range gm {
+			tk, v = unwrapValue(v)
+			switch strings.ToLower(k) {
+			case "name":
```

Comment:
ok so this is the other cluster. We should try to make this more concrete and clear. I think they have to be the same,  clustername I mean.